### PR TITLE
Support model class inheritance. Fixes #164

### LINKED ISSUE #164: Inheriting from Model - should this work?
Hi,

Should it be possible to sub-class a pynamodb Model?  I'm getting the following error:

```
AttributeError: As of v1.0 PynamoDB Models require a `Meta` class with a table_name property.
                See https://pynamodb.readthedocs.io/en/latest/release_notes.html
```

MRE to reproduce (Python 2.7.12, pynamodb-1.5.3):

``` python
from pynamodb.models import Model
from pynamodb.attributes import UnicodeAttribute, NumberAttribute

class SomeBase(Model):

    class Meta:
        table_name='table-one'
        region     = 'us-west-2'

    email = UnicodeAttribute(hash_key=True)
    count = NumberAttribute(default=0)


class VariantA(SomeBase):

    def __init__(self, *args, **kwargs):
        super(VariantA, self).__init__(*args, **kwargs)


if __name__ == '__main__':
    some_a = VariantA()

    #print "Meta attribute present: {}".format(hasattr(some_a, 'Meta'))
    #print dir(some_a)

    some_a.email = 'user@example.com'
    some_a.count = 666
    some_a.save()
```

### PR DESCRIPTION
**Pull request by jpinner-lyft**
`DefaultMeta` is removed so that a model's Meta class is inherited from the parent class.

Note: this is technically a backwards-incompatible change for those users who relied on dynamically setting attributes of their Meta class without creating it first.

### CODE REVIEW

**Comment by jpinner-lyft** [comment]
reattempt of https://github.com/pynamodb/PynamoDB/pull/412

File: pynamodb/models.py [hunk 1]
```diff
@@ -151,10 +151,6 @@ def commit(self) -> None:
             unprocessed_items = data.get(UNPROCESSED_ITEMS, {}).get(self.model.Meta.table_name)
 
 
-class DefaultMeta(object):
```
Comment (jpinner-lyft) [inline pynamodb/models.py:154]: this is leftover from when default values were stored in this object

**Comment by jpinner-lyft** [comment]
Question from the previous attempt: say I have

```
ModelA(Model):
    class Meta:
        table_name = 'foo'

ModelB(ModelA):
    pass
```
with this change, ModelB.Meta.table_name == 'foo'

Now if i do ModelA.Meta.table_name = 'bar' what should ModelB.Meta.table_name be?
Or vice-versa, if i do ModelB.Meta.table_name = 'bar' should that change ModelA?

Do I need to make Meta some sort of "inherited, copy-on-write-per-property" class?

My current thoughts on this - subclassing will be initially used to support polymorphism where the parent model and its children classes all point to the same table so modifying the Meta class on any of the classes should modify it on all of them (they all reference the same configuration).

**Comment by jpinner-lyft** [comment]
Note: this PR does not fix #361 since connections are still cached on the Model object. A follow-up change to cache connections on the Meta class will address that issue.

**Review by garrettheel — APPROVED** [review]

**Pull request merged**